s2sml/torch_funcs.py

```diff
@@ -732,16 +732,23 @@ def corrcoef(pred, target):
     # https://forum.numer.ai/t/custom-loss-functions-for-xgboost-using-pytorch/960
     
     """
-    pred[pred==1] = 0.999
-    pred[pred==-1] = -0.999
-
     pred_n = pred - pred.mean()
     target_n = target - target.mean()
-    pred_n = pred_n / pred_n.norm()
+
+    pred_norm = pred_n.norm() 
+
+    # if predictions are constant, norm would be zero, and correlation would be infinitely small, corrected by 0.0001
+    if pred_norm == 0: 
+        val_to_return = pred_norm.clone()
```

Comment:
Should we be cloning pred_n instead of pred_norm? @jhayron-perez

Reply:
If the shapes stay consistent, then it doesn't matter

Reply:
Oh wait, this actually works because we are returning the coefficient, so it needs to be a single value (e.g., the norm)